pkg/apis/monitoring/v1/types.go

```diff
@@ -1626,3 +1640,58 @@ func (l *PrometheusRuleList) DeepCopyObject() runtime.Object {
 type ProbeTLSConfig struct {
 	SafeTLSConfig `json:",inline"`
 }
+
+// AuthTLSConfig specifies TLS configuration parameters.
+// +k8s:openapi-gen=true
+type AuthTLSConfig struct {
+	SafeTLSConfig `json:",inline"`
+}
+
+// AuthCredentials allows to specify authorization credentials as either
+// BearerToken, BasicAuth, OAuth or TLS certificate. Fields are mutually exclusive.
+type AuthCredentials struct {
```

Comment:
My impression is that this type is redundant. The `authorization` section in the prometheus config is described as follows:
```
# Sets the `Authorization` header on every scrape request with the configured credentials.
```
It is not intended to be used as generic authorization, but rather configures the `Authorization` header value.

Reply:
The `Authorization` header supports a number of [auth schemes](https://www.iana.org/assignments/http-authschemes/http-authschemes.xhtml). Prometheus afaiu should support at least Bearer, Basic, OAuth. TLS too but I'm not sure if that applies to the `Authorization` header.